cmake/modules/platform/toolcfg/gnu.cmake

```diff
@@ -88,21 +88,27 @@ endif()
 function(_omr_toolchain_separate_debug_symbols tgt)
 	set(exe_file "$<TARGET_FILE:${tgt}>")
 	if(OMR_OS_OSX)
-		set(dbg_file "${exe_file}.dSYM")
+		set(dbg_file "${exe_file}${OMR_DEBUG_INFO_OUTPUT_EXTENSION}")
+		if(OMR_FLATTEN_DEBUG_INFO)
+			set(flatten_arg "-f")
+		else()
+			set(flatten_arg)
+		endif()
 		add_custom_command(
 			TARGET "${tgt}"
 			POST_BUILD
-			COMMAND dsymutil -o "${dbg_file}" "${exe_file}"
+			COMMAND dsymutil ${flatten_arg} -o "${dbg_file}" "${exe_file}"
 		)
 	else()
-		omr_get_target_path(target_path ${tgt})
-		omr_replace_suffix(dbg_file "${target_path}" ".debuginfo")
+		omr_get_target_output_info(${tgt} output_dir output_name)
+		set(dbg_filename "${output_name}${OMR_DEBUG_INFO_OUTPUT_EXTENSION}")
+		set(dbg_file "${output_dir}/${dbg_filename}")
 		add_custom_command(
 			TARGET "${tgt}"
 			POST_BUILD
 			COMMAND "${CMAKE_OBJCOPY}" --only-keep-debug "${exe_file}" "${dbg_file}"
 			COMMAND "${CMAKE_OBJCOPY}" --strip-debug "${exe_file}"
-			COMMAND "${CMAKE_OBJCOPY}" --add-gnu-debuglink="${dbg_file}" "${exe_file}"
+			COMMAND "${CMAKE_OBJCOPY}" --add-gnu-debuglink="${dbg_filename}" "${exe_file}"
```

Comment:
This now fails because `dbg_filename` is a relative path that's not meaningful in the working directory.

Reply:
Hmm the thought was that it should be a relative path (as I believe the full path you provide is embedded into the debug link, which would mean that debug info no longer works once you move the debug info. Will try setting the working directory, which should fix the issue

Reply:
Just played around with it, and as it turns out, even if you specify a full path the embedded debug link only keeps the filename. Will revert to the old behavior